x-pack/plugins/session_view/public/components/process_tree/helpers.ts

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { Process, ProcessEvent, ProcessMap } from '../../../common/types/process_tree';
+import { ProcessImpl } from './hooks';
+
+export const updateProcessMap = (processMap: ProcessMap, events: ProcessEvent[]) => {
```

Comment:
It might make sense to add comments to the helpers in this file that explain what they do and that they are mutating objects.